include/lbann/proto/factories.hpp

```diff
@@ -49,17 +49,23 @@ class Weights;
 namespace lbann {
 namespace proto {
 
+/** Construct a trainer specified with a prototext. */
+std::unique_ptr<trainer> construct_trainer(lbann_comm* comm,
```

Comment:
What's a `trainer`? Missing a `#include` (or several, but those aren't specific to this PR...)